SUASSystem/SUASSystem/image_processing.py

```diff
@@ -1,17 +1,51 @@
 from time import sleep
-import os
-from SUASSystem import utils
+import os, math
+from SUASSystem import utils, GCSSettings, inverse_haversine
+from PIL import Image
 
 def run_img_proc_process(logger_queue, img_proc_status, location_log, targets_to_submit, interop_client_array):
+
     while True:
+
         if len(targets_to_submit) > 0:
             target_characteristics=targets_to_submit.pop(0)
+
+            target_time=utils.get_image_timestamp(target_characteristics["base_image_filename"])
+
+            closest_time_index=0
+            least_difference = location_log[0]["epoch_time"]
+            for index in range(len(location_log)):
+                difference = target_time-location_log[closest_time_index]["epoch_time"]
+                if(abs(difference) <= least_difference):
```

Comment:
If statements do not need parentheses around them. For instance, this should be:

```if abs(difference) <= least_difference:```